lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1430,29 +1431,17 @@ bool WitnessChecker::findBestWitness(
 }
 
 bool WitnessChecker::
-checkWitnessAccessibility(const DeclContext *&requiredAccessScope,
+checkWitnessAccessibility(AccessScope &requiredAccessScope,
                           ValueDecl *requirement,
                           ValueDecl *witness,
                           bool *isSetter) {
   *isSetter = false;
 
-  const DeclContext *protoAccessScope = Proto->getFormalAccessScope(DC);
+  requiredAccessScope =
+    *requiredAccessScope.intersectWith(Proto->getFormalAccessScope(DC));
```

Comment:
Ah, right, my bad. (Somehow I misread this as equivalent to `requiredAccessScope->intersectWith`.) The explicit assert is probably a good idea just to make it a little faster to track down what happened if it ever breaks.